### DISCUSSION TITLE
✅ Keyboard not showing back up

### DISCUSSION THREAD

**Discussion by githubJMB88** [isontheline/pro.webssh.net#903] [General]
Is it just me or is there a problem with the keyboard not showing back up after minimizing it or opening a new tab?

It's been like this with me for the past week or two (2)

**Comment by isontheline** [comment]
@githubJMB88 Sorry I don't understand. Could you explain me further?

Thank you

**Comment by djralph** [comment]
Hi. I have the same issue. When I click on the screen terminal, keyboard is hiding and I can’t „unhide” it…

**Reply by isontheline** [reply]
You should tap the terminal briefly : less than < 0.1 sec
<img width="228" alt="image" src="https://github.com/isontheline/pro.webssh.net/assets/44212923/6db3f2a3-187b-4462-9ad7-1750553a1b02">

In order to show the keyboard again

**Reply by djralph** [reply]
Doesn’t work.

Iphone XS Max
iOS 16.5

**Reply by isontheline** [reply]
Even if you don't move the finger while release it from the screen?

**Reply by djralph** [reply]
Trust me, I tried every kind of tap ;)
Hiding works perfect - one tap on terminal, but another second tap doesn’t show the keyboard.

**Reply by isontheline** [reply]
I trust you 😉

Could you enable verbose logging while you try again?
I would like to find some root cause.

In order to enable verbose logging you can follow the steps :
1. iOS Settings
2. WebSSH
3. Settings
4. Log Level > Verbose
5. Restart WebSSH
6. Reproduce your action
7. Find the log files inside iOS > Files > WebSSH

**Reply by djralph** [reply]
Log attached.
I taped many, many times in time the log was enabled. 
[log-43C6AA.txt](https://github.com/isontheline/pro.webssh.net/files/11736855/log-43C6AA.txt)

**Reply by isontheline** [reply]
Thank you @djralph 🙏

What about the latest beta release?

https://testflight.apple.com/join/QSrBK59z

Does the issue is also encountered?

**Reply by djralph** [reply]
The same…

**Reply by isontheline** [reply]
I'm opening an issue to track this trouble : https://github.com/isontheline/pro.webssh.net/issues/917

I will release a new beta (in a few minutes/hours) to add more logs in order to find the root cause

**Comment by djralph** [comment]
Watch this.
https://1drv.ms/v/s!AioZQ9w1H4MyiIp_qVIbE6WVCIguYA

**Reply by isontheline** [reply]
Thank you so much for this video 🙏

Could you extract the logs by using WebSSH 22.3.1112?
https://testflight.apple.com/join/QSrBK59z

**Reply by djralph** [reply]
[log-43C6AA.txt](https://github.com/isontheline/pro.webssh.net/files/11749801/log-43C6AA.txt)

**Reply by isontheline** [reply]
Thank you!

May you now try 22.3.1113?

**Comment by djralph** [comment]
Same :(
Before I hide the keyboard, the cursor is outlined and filled green. When hidden, the cursor is only outlined. Maybe you should check it out.

[log-43C6AA.txt](https://github.com/isontheline/pro.webssh.net/files/11750069/log-43C6AA.txt)

**Reply by isontheline** [reply]
That's so strange as I unable to reproduce on my device nor on the simulator.
But I will find why!

**Reply by djralph** [reply]
I checked on my girlfriend’s IPhone 11 Pro and it works well. Maybe the screen size/resolution matters. I have IPhone XS Max (the bigger one).

**Reply by isontheline** [reply]
Thanks for your feedback @djralph 🙏

Could you try the latest beta release? 22.3.1114
I have added som delays

**Comment by djralph** [comment]
Same…

**Reply by isontheline** [reply]
@githubJMB88 Have you some time to try 22.3.1115 and extract logs?

**Reply by djralph** [reply]
Now is better, but still not too good. To unhide the keyboard I have to tap few times and it show up.

**Reply by djralph** [reply]
[log-43C6AA.txt](https://github.com/isontheline/pro.webssh.net/files/11759017/log-43C6AA.txt)

**Reply by isontheline** [reply]
Thank you!

It's hard for me to fix it as I can't reproduce it on any device.
But it's a true issue as you are several to reports this glitch.

I'm searching if it can comes from an iOS setting that other users don't have enabled.

**Reply by djralph** [reply]
Maybe try to find someone with same IPhone model - Xs Max. I’m not sure it is not my device issue, but I don’t have problems with any other apps.

**Reply by isontheline** [reply]
As I have made a "big" change inside 22.3.1116 release, I don't release it to all testers.
I have updated how I handle events but not sure if it would help as much as needed.

To download it, please use this link instead :
https://testflight.apple.com/join/LwlfYbRH

Could you give it a try?

**Reply by djralph** [reply]
I can’t see this version on TestFlight

**Reply by isontheline** [reply]
I don't see you on this new link -> https://testflight.apple.com/join/LwlfYbRH

**Reply by isontheline** [reply]
@djralph I've just pushed a new beta release :
<img width="386" alt="image" src="https://github.com/isontheline/pro.webssh.net/assets/44212923/ac4475fe-f8cd-480b-ad24-3b5d9af9b4d5">

A new keyboard icon will appear at top.

Could you try to use it?

**Reply by lifejw** [reply]
Keyboard icon works 👌🏼
Keyboard reappears and can also be hidden.

**Reply by isontheline** [reply]
@lifejw Thank you so much for your feedback Jonathan 🙏

**Reply by djralph** [reply]
Ok. Icon works well, I can hide and unhide keyboard, but it is strange to mark the words and paste from clipboard. Maybe this is connected with unhiding keyboard issue… Watch the video 
https://1drv.ms/v/s!AioZQ9w1H4MyiIsNEHZaN4B13sS2UA

**Reply by isontheline** [reply]
@djralph Yes I have other glitches on text selection like this : https://github.com/isontheline/pro.webssh.net/issues/909

These will be addressed asap
